src/portable/portable.c

```diff
@@ -713,11 +785,16 @@ static int install_chroot_dropin(
                 if (!strextend(&text,
                                "\n"
                                "[Service]\n",
-                               IN_SET(type, IMAGE_DIRECTORY, IMAGE_SUBVOLUME) ? "RootDirectory=" : "RootImage=", image_path, "\n"
-                               "Environment=PORTABLE=", basename(image_path), "\n"
+                               root_type, image_path, "\n"
+                               "Environment=PORTABLE=", basename(m->image ?: image_path), "\n"
                                "BindReadOnlyPaths=", os_release_source, ":/run/host/os-release\n"
-                               "LogExtraFields=PORTABLE=", basename(image_path), "\n"))
+                               "LogExtraFields=PORTABLE=", basename(m->image ?: image_path), "\n"))
```

Comment:
given the expression in the basename() is getting more complex and complex let's prepare that ahead of time,and reuse it twice

i.e. add 

```c
r = path_extract_filename(m->image ?: image_path, &base_name);
…
```

or so, and then use the extracted value everywhere